ts/tabSwitch.ts

```diff
@@ -21,33 +23,37 @@ function setActiveTabListener() {
 }
 
 function toggleTodosVisibility(
-  trashCansVisibility,
-  deleteCompletedButtonDisplay,
-  taskAddButtonDisplay
+  trashCansVisibility: string,
+  deleteCompletedButtonDisplay: string,
+  taskAddButtonDisplay: string
 ) {
-  let trashCans = document.querySelectorAll(".completed-trash");
-  trashCans.forEach((e) => (e.style.visibility = trashCansVisibility));
+  let trashCans = document.querySelectorAll(
+    ".completed-trash"
+  ) as NodeListOf<HTMLElement>;
+  trashCans.forEach((e) => {
+    e.style.visibility = trashCansVisibility;
+  });
   deleteCompletedButton.style.display = deleteCompletedButtonDisplay;
   taskAddButton.style.display = taskAddButtonDisplay;
 }
 
-function updateTodoDisplay(todo, displayValue) {
+function updateTodoDisplay(todo: HTMLElement, displayValue: string) {
   todo.style.display = displayValue;
 }
 
 function showAllTodos() {
   toggleTodosVisibility("hidden", "none", "flex");
-  const todos = reassignTodos()[0];
+  const todos = reassignTodos()[0] as NodeListOf<HTMLElement>;
   todos.forEach((todo) => {
     updateTodoDisplay(todo, "block");
   });
 }
 
 function showActiveTodos() {
   toggleTodosVisibility("hidden", "none", "flex");
```

Comment:
This is weird to be honest. First of all you repeat this call 2 times, in active and all. Second, why not just pass to this function the value of the selected tab ? If it's `all` of `active`, you do this logic, if not, you do `visible`, `block` and `noone`